### ISSUE TITLE
[QA] Cycle 2: Feature testing for reCAPCTHA

### ISSUE THREAD

**Issue by sdzhepa** [magento/securify#201]
## Code version information:
On the starting day we used next commits and branch
- Magento Open Source: branch `2.4-develop` commit:  https://github.com/magento/magento2/commit/15dc8749fdb88bff5875d6305a621532ad4a966c
- Magento Commerce(EE): branch `2.4-develop` commit: https://github.com/magento/magento2ee/commit/e736a90af06591661ba1653436a300d7674b5218
- Magento security-package: branch `1.0-develop` commit: https://github.com/magento/security-package/commit/496055e9996ddeac586acfff7262f7ca443f45d5

	NOTE: _IT is possible that during execution codebase could be updated due to critical issues that block test execution. Or exceptional situation_
	
## Magento editions for test-case execution:

**1. Magento Commerce(EE)**
-- The main env for full testing
-- Prepared Cloud instance with sample data
-- Should be executed all test-cases that we have. (exception could be only for specific cases that not possible to execute due to Cloud Env)

**2. Magento Open Source(CE)**
-- Additional env for execution(on local machine). Install using [Metapackage Installation Guide](https://github.com/magento/security-package/wiki/Metapackage-Installation-Guide)
-- Should contains only the most critical cases e.g. Checkout Flow etc
-- Here should be executed tests that not possible to do on EE due to limitation on Cloud env
-- All found issues on EE should be checked on CE before reporting

### Should start in the next order:
1. [Cycle related to Magento Commerce(EE) Cloud.](https://studio.cucumber.io/projects/169934/test-runs/390580/overview)
2. [Cycle related to Magento Open Source(CE) Local](https://studio.cucumber.io/projects/169934/test-runs/390585/overview)
3. [Cycle related to cases/user flows in different browser](https://studio.cucumber.io/projects/169934/test-runs/390594/overview)

### Bug reporting:
- Each found bug should be reverified on CE local env. To be sure is it env/edition specific or not
- Each bug should be reported as an issue in this repository. It should have labels: 'Component: Google reCAPCTHA' and 'bug'

### Failed/Skipped/Blocked test cases:
- Each FAILED test-case should have a link to the created issue in the "Execution" section and additional comment if required
- Each SKIPPED/BLOCKED test-case should have a clear comment with explanation in the "Execution" section

### Bug triaging and fixing
- Each newly created bug should be triaged. Discuss and understand its priority/severity. Must be fixed now or can be postponed
- Only after triage and decision, devs can start progress on it.
- Each fixed bug must be tested in the scope of Pull Request.
- Based on the number of new bugs and specific will be decided do we need additional Test Run or not

### Magento Commerce Cloud

Name: [Cycle 2: reCAPCTHA testing on Magneto Cloud](https://studio.cucumber.io/projects/169934/test-runs/390580/overview)

_Additional testing scope:_
- [ ]   https://github.com/magento/security-package/issues/102
- [ ] https://github.com/magento/security-package/pull/191
- [ ] https://github.com/magento/security-package/pull/183
- [ ] https://github.com/magento/security-package/pull/174

### Magento Open Source(CE) Local

Name: [Cycle 2: reCAPCTHA testing on Magneto Open Source](https://studio.cucumber.io/projects/169934/test-runs/390585/overview)

_Additional testing scope:_
- [ ] upgrade https://github.com/magento/security-package/pull/89
- [ ] cli commands to disable reCAPCTH for admin https://github.com/magento/security-package/pull/147
- [ ] https://github.com/magento/security-package/pull/175 
- [ ] check system/exception log files during execution, is it has enny errors/warnigs/etc

### Browser specific testing (Google Chrome, Mozilla Firefox, IE11/Edge)
Name: [Cycle 2: reCAPCTHA testing on Magneto Open Source](https://studio.cucumber.io/projects/169934/test-runs/390594/overview)

_Additional testing scope:_
- [ ] https://github.com/magento/security-package/pull/190#issuecomment-608406917

**Comment by sdzhepa** [comment]
Testing report on Apr 10 2020

Start testing.
**No new bugs found**



Details
1. [Cycle related to Magento Commerce(EE) Cloud.](https://studio.cucumber.io/projects/169934/test-runs/390580/overview)
- Total test cases: 70
- Passed: 20
- Failed: 0
- Skipped 0

2. [Cycle related to Magento Open Source(CE) Local](https://studio.cucumber.io/projects/169934/test-runs/390585/overview)
- Total test cases: 38
- Passed: 11
- Failed: 0
- Skipped 0

3. [Cycle related to cases/user flows in different browser](https://studio.cucumber.io/projects/169934/test-runs/390594/overview)
- Total test cases: 21
- Passed: 0
- Failed: 0
- Skipped 0

**Comment by sdzhepa** [comment]
Testing report on Apr 14 2020 (3rd day of testing)

**Found Bugs**
- https://github.com/magento/security-package/issues/204
- https://github.com/magento/security-package/issues/203
- https://github.com/magento/security-package/issues/205


Details
1. [Cycle related to Magento Commerce(EE) Cloud.](https://studio.cucumber.io/projects/169934/test-runs/390580/overview)
- Total test cases: 70
- Passed: 51
- Failed: 2
- Skipped: 5 [List](https://studio.cucumber.io/projects/169934/test-runs/390580/overview)

2. [Cycle related to Magento Open Source(CE) Local](https://studio.cucumber.io/projects/169934/test-runs/390585/overview)
- Total test cases: 38
- Passed: 21
- Failed: 0
- Skipped 0

3. [Cycle related to cases/user flows in different browser](https://studio.cucumber.io/projects/169934/test-runs/390594/overview)
- Total test cases: 21 per each browser(FireFox, Chrome, IE11, Endge)

**FireFox**
- Passed: 0
- Failed: 0

**Chrome**
- Passed: 13
- Failed: 0

**IE 11**
- Passed: 0
- Failed: 0

**Edge**
- Passed: 0
- Failed: 0

**Comment by sdzhepa** [comment]
Testing report on Apr 15, 2020 (4rd day of testing)

No new bugs found

**All Found Bugs**
- https://github.com/magento/security-package/issues/204
- https://github.com/magento/security-package/issues/203
- https://github.com/magento/security-package/issues/205


Details
1. [Cycle related to Magento Commerce(EE) Cloud.](https://studio.cucumber.io/projects/169934/test-runs/390580/overview)
- Total test cases: 70
- Passed: 56
- Failed: 3
- Skipped: 7 [List](https://studio.cucumber.io/projects/169934/test-runs/390580/overview)

2. **DONE** [Cycle related to Magento Open Source(CE) Local](https://studio.cucumber.io/projects/169934/test-runs/390585/overview)
- Total test cases: 38
- Passed: 38
- Failed: 0
- Skipped 0

3. [Cycle related to cases/user flows in different browser](https://studio.cucumber.io/projects/169934/test-runs/390594/overview)
- Total test cases: 21 per each browser(Firefox, Chrome, IE11, Edge)

**FireFox**
- Passed: 14
- Failed: 0

**Chrome**
- Passed: 14
- Failed: 0

**IE 11**
- Passed: 0
- Failed: 0

**Edge**
- Passed: 0
- Failed: 0

**Comment by engcom-Delta** [comment]
Results for addional testing scope:
### Magento Commerce Cloud

Name: [Cycle 2: reCAPCTHA testing on Magneto Cloud](https://studio.cucumber.io/projects/169934/test-runs/390580/overview)

_Additional testing scope:_
- [x]   https://github.com/magento/security-package/issues/102 - **Done**
- [x] https://github.com/magento/security-package/pull/191 - **Related issue https://github.com/magento/security-package/issues/205 according [comment](https://github.com/magento/security-package/pull/175#issuecomment-610588582)**
- [x] https://github.com/magento/security-package/pull/183 - **Done**
- [x] https://github.com/magento/security-package/pull/174 - **no longer valid due to https://github.com/magento/security-package/pull/175**

### Magento Open Source(CE) Local

Name: [Cycle 2: reCAPCTHA testing on Magneto Open Source](https://studio.cucumber.io/projects/169934/test-runs/390585/overview)

_Additional testing scope:_
- [x] upgrade https://github.com/magento/security-package/pull/89 - **Done**
- [x] cli commands to disable reCAPCTH for admin https://github.com/magento/security-package/pull/147 - **Done**
- [x] https://github.com/magento/security-package/pull/175 - **Related issue https://github.com/magento/security-package/issues/205 for better error handler for v2-v3 Invisible: not fixed by https://github.com/magento/security-package/pull/191 (Note: Initial issue is fixed)**
- [x] check system/exception log files during execution, is it has enny errors/warnigs/etc - **Done: no additional errors was found**


### Browser specific testing (Google Chrome, Mozilla Firefox, IE11/Edge)
Name: [Cycle 2: reCAPCTHA testing on Magneto Open Source](https://studio.cucumber.io/projects/169934/test-runs/390594/overview)

_Additional testing scope:_
- [x] https://github.com/magento/security-package/pull/190#issuecomment-608406917 - **Done**

**Comment by sdzhepa** [comment]
Testing report on Apr 16, 2020 (5th day of testing)

All planned test cycles were executed.
Testing is done

**All Found Bugs**
- https://github.com/magento/security-package/issues/204
- https://github.com/magento/security-package/issues/203
- https://github.com/magento/security-package/issues/205
- https://github.com/magento/security-package/issues/207 (not regression bug and not related to reCACPCTHA)

Details
1. **DONE** [Cycle related to Magento Commerce(EE) Cloud.](https://studio.cucumber.io/projects/169934/test-runs/390580/overview)
- Total test cases: 70
- Passed: 60
- Failed: 3
- Skipped: 7 [List](https://studio.cucumber.io/projects/169934/test-runs/390580/overview)

2. **DONE** [Cycle related to Magento Open Source(CE) Local](https://studio.cucumber.io/projects/169934/test-runs/390585/overview)
- Total test cases: 38
- Passed: 38
- Failed: 0
- Skipped 0

3. [Cycle related to cases/user flows in different browser](https://studio.cucumber.io/projects/169934/test-runs/390594/overview)
- Total test cases: 21 per each browser(Firefox, Chrome, IE11, Edge)

**FireFox**
- Passed: 14
- Failed: 0

**Chrome**
- Passed: 14
- Failed: 0

**IE 11**
- Passed: 14
- Failed: 0

**Edge**
- Passed: 14
- Failed: 0


4. **DONE** Addition testing scope
- https://github.com/magento/security-package/issues/201#issuecomment-614629904
